src/main/java/executor/service/service/ExecutionService.java

```diff
@@ -1,18 +1,87 @@
+import org.openqa.selenium.WebDriver;
+
+import java.util.Queue;
+
+import static executor.service.config.properties.PropertiesConstants.*;
+
+/**
+ * The facade for execute ScenarioExecutor.
+ *
+ *  @author Oleksandr Tuleninov
+ *  @version 01
+ * */
 public class ExecutionService {
 
-    private WebDriverInitializer driverInitializer;
     private ScenarioExecutor scenarioExecutor;
-    private ScenarioSourceListener scenarioSourceListener;
+    private PropertiesConfig propertiesConfig;
+    private WebDriverConfig webDriverConfig;
 
-    public ExecutionService(WebDriverInitializer driverInitializer, ScenarioExecutor scenarioExecutor, ScenarioSourceListener scenarioSourceListener) {
-        this.driverInitializer = driverInitializer;
+    public ExecutionService() {
+    }
+
+    public ExecutionService(ScenarioExecutor scenarioExecutor,
+                            PropertiesConfig propertiesConfig,
+                            WebDriverConfig webDriverConfig) {
         this.scenarioExecutor = scenarioExecutor;
-        this.scenarioSourceListener = scenarioSourceListener;
+        this.propertiesConfig = propertiesConfig;
+        this.webDriverConfig = webDriverConfig;
+    }
+
+    /**
+     * Execute ScenarioExecutor.
+     *
+     * @param scenarios the queue with scenarios
+     * @param proxies   the queue with proxies
+     */
+    public void execute(Queue<Scenario> scenarios, Queue<ProxyConfigHolder> proxies) {
+        WebDriverConfig configuredWebDriverConfig = configureWebDriverConfig(propertiesConfig, webDriverConfig);
+
+        for (Scenario scenario : scenarios) {
+            ProxyConfigHolder proxy = proxies.poll();
+            if (scenario == null || proxy == null) continue;
+
+            WebDriver webDriver = getWebDriverPrototype(configuredWebDriverConfig, proxy);
+            if (webDriver == null) continue;
+
+            scenarioExecutor.execute(scenario, webDriver);
+        }
+    }
+
+    /**
+     * Configure WebDriverConfig from properties file.
+     *
+     * @param propertiesConfig the properties from resources file
+     * @param webDriverConfig  the WebDriverConfig entity
+     */
+    private WebDriverConfig configureWebDriverConfig(PropertiesConfig propertiesConfig, WebDriverConfig webDriverConfig) {
+        var properties = propertiesConfig.getProperties(WEB_DRIVER);
+        var webDriverExecutable = properties.getProperty(WEB_DRIVER_EXECUTABLE);
+        var userAgent = properties.getProperty(USER_AGENT);
+        var pageLoadTimeout = Long.parseLong(properties.getProperty(PAGE_LOAD_TIMEOUT));
+        var implicitlyWait = Long.parseLong(properties.getProperty(IMPLICITLY_WAIT));
+        webDriverConfig.setWebDriverExecutable(webDriverExecutable);
+        webDriverConfig.setUserAgent(userAgent);
+        webDriverConfig.setPageLoadTimeout(pageLoadTimeout);
+        webDriverConfig.setImplicitlyWait(implicitlyWait);
+
+        return webDriverConfig;
```

Comment:
this method looks pretty dirty and weird. why this function takes *webDriverConfig* as a parameter? is given *webDriverConfig* empty or has some values? why just not to use static builder for this class? why just not to make this *WebDriverConfig* readonly singletone, setup all the constants there and inject this class?

Reply:
This WebDriverConfig injects DI as a singleton on the default constructor and is configured in the method.